File: 1657-determine-if-two-strings-are-close/1657-determine-if-two-strings-are-close.cpp
class Solution {
public:
    bool closeStrings(string word1, string word2) {
        if(word1.size() != word2.size()) return false;
        int w1[26];
        int w2[26];
        memset(w1, 0, sizeof(w1));
        memset(w2, 0, sizeof(w2));
        for(auto ch : word1) {
            w1[ch-'a'] += 1;
        }
        for(auto ch : word2) {
            w2[ch-'a'] += 1;
        }
        for(int i=0; i<26; i++) {
            if(w1[i] != 0 && w2[i] == 0) return false;
        }
        sort(w1, w1 + sizeof(w1)/sizeof(w1[0]));
        sort(w2, w2 + sizeof(w2)/sizeof(w2[0]));
        for(int i=0; i<26; i++) {
            if(w1[i] != w2[i]) return false;
        }
        return true;
    }
};

File: 1657-determine-if-two-strings-are-close/1657-determine-if-two-strings-are-close.py
class Solution:
    def closeStrings(self, word1: str, word2: str) -> bool:
        if len(word1) != len(word2): return False
        w1 = [0 for i in range(26)]
        w2 = [0 for i in range(26)]
        for ch in word1:
            w1[ord(ch) - ord('a')] += 1
        for ch in word2:
            w2[ord(ch) - ord('a')] += 1
        for i in range(26):
            if w1[i] == 0 and w2[i] != 0: return False
        w1.sort()
        w2.sort()
        for i in range(26):
            if w1[i] != w2[i]: return False
        return True